extensions/wikia/CommunityHeader/CommunityHeaderHooks.class.php

```diff
@@ -7,4 +7,28 @@ public static function onBeforePageDisplay( /*\OutputPage $out, \Skin $skin */ )
 
 		return true;
 	}
+
+
+	/**
+	 * Render the preview of wiki navigation menu
+	 *
+	 * @param Title $title Title of the page preview is generated for
+	 * @param string $html preview content to modify
+	 * @param string $wikitext current wikitext from the editor
+	 * @return bool return true
+	 */
+	public static function onEditPageLayoutModifyPreview( Title $title, &$html, $wikitext ) {
+		if ( NavigationModel::isWikiNavMessage( $title ) ) {
+			// render a preview
+			$html =
+				F::app()
+					->renderPartial( 'CommunityHeader', 'localNavigationPreview',
+						[ 'wikiText' => $wikitext ] );
+
+			// open links in new tab
+			$html = str_replace( '<a ', '<a target="_blank" ', $html );
```

Comment:
maybe a parameter to localNavigation preview - to already generate links with target?